File: GC.Domain/Users/SystemUser.cs
using System;

namespace GC.Domain.Users
{
    public class SystemUser
    {
        public Guid Id { get; }
        public String Login { get; }
        public UserPermission Permission { get; }
        public Boolean HasOtherPermissions { get; }
        public String[] AvailableAccessPolicies { get; }

        public SystemUser(User user, UserPermission permission, Boolean hasOtherPermissions, String[] availableAccessPolicies)
        {
            Id = user.Id;
            Login = user.Login;
            Permission = permission;
            HasOtherPermissions = hasOtherPermissions;
            AvailableAccessPolicies = availableAccessPolicies;
        }
    }
}

File: GC.WebSpace/ClientApp/domain/users/systemUser.ts
import { AccessPolicy } from "../accesspolicies/accessPolicy";
import { UserPermission } from "./userPermission";

export default class SystemUser {
    public static id: string;
    public static login: string;
    public static permission: UserPermission;
    public static hasOtherPermissions: boolean;
    public static availableAccessPolicies: string[]

    public static loadSystemUser() {
        const systemUser = (window as any).systemUser;

        if (systemUser != null) {
            SystemUser.id = systemUser.id;
            SystemUser.login = systemUser.login;
            SystemUser.permission = systemUser.permission;
            SystemUser.hasOtherPermissions = systemUser.hasOtherPermissions;
            SystemUser.availableAccessPolicies = systemUser.availableAccessPolicies;
        }
    }

    public static hasAccess = (accessPolicy: AccessPolicy) => {
        return SystemUser.availableAccessPolicies.includes(accessPolicy);
    }
}

SystemUser.loadSystemUser();
